src/cache/precomputed_quantities.jl

```diff
@@ -480,8 +480,10 @@ NVTX.@annotate function set_implicit_precomputed_quantities_part1!(Y, p, t)
 
     # TODO: We might want to move this to dss! (and rename dss! to something
     # like enforce_constraints!).
-    set_velocity_at_surface!(Y, ᶠuₕ³, turbconv_model)
-    set_velocity_at_top!(Y, turbconv_model)
+    if !(p.atmos.prescribed_flow isa PrescribedFlow)
+        set_velocity_at_surface!(Y, ᶠuₕ³, turbconv_model)
+        set_velocity_at_top!(Y, turbconv_model)
+    end
```

Comment:
In simulations with topography, `set_velocity_at_surface!` and `set_velocity_at_top!` are needed to ensure that the state's velocity components match its advective tendency. If the advective tendency assumes zero-flux (`CT3(0)`) boundary conditions, these functions modify `Y.f.u₃` to enforce that. In your case, fluxes at the surface are prescribed some nonzero value (i.e., the projection `CT3(ρ_sfc * q_tot_sfc * u₃_sfc)`, where `u₃_sfc` is specified as a `WVector`), so `set_velocity_at_surface!` and `set_velocity_at_top!` should make `Y.c.uₕ` and `Y.f.u₃` consistent with that.

If you never intend to run this configuration with topography, then you can leave this as is, but add a check somewhere in `type_getters.jl` to make sure that there is no topography.

Reply:
Leaving this as-is for now, and added check to `type_getters.jl`